### PR TITLE
License

### LINKED ISSUE #22: Add license header to all files
- Add [licensei](https://github.com/goph/licensei) to the tooling and CI to verify headers (example: https://github.com/bank-vaults/vault-secrets-webhook)
- Add license headers to Go files ([addlicense](https://github.com/google/addlicense) can make it easier for the first time)

Copyright holder is `Tailfin Cloud Inc.`.

### PR DESCRIPTION
**Pull request by KunalSin9h**
## Overview

Fixes #22 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .licensei.toml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+approved = [
+  "mit",
+  "apache-2.0",
+  "bsd-3-clause",
+  "bsd-2-clause",
+  "mpl-2.0",
+]
+
+[header]
+ignorePaths = ["vendor", "client", ".gen"]
```
Comment (sagikazarmark) [inline .licensei.toml:10]:
```suggestion
ignorePaths = [".devenv", ".direnv", "vendor", "client", ".gen"]
```

**Review by sagikazarmark — CHANGES_REQUESTED** [review]
Good work, thanks @KunalSin9h! Can you please take a look at that one change request? Thanks!

**Comment by KunalSin9h** [comment]
@sagikazarmark can you help here, it is failing in the license-check CI, saying `licensi` not found!

**Comment by KunalSin9h** [comment]
i think after editing `flake.nix` i have not updated the `flake.lock` file, this might be a problem for this.

**Comment by sagikazarmark** [comment]
@KunalSin9h I added licensei for you in #74

Please rebase this branch instead of merging main into it.

**Comment by KunalSin9h** [comment]
i think i have to create a new branch for this.

**Pull request closed without merging**